src/fides/api/api/v1/endpoints/generic_overrides.py

```diff
@@ -124,29 +124,36 @@ async def create_data_use(
     """
     Create a data use. Updates existing data use if data use with name already exists and is disabled.
     """
-    if data_use.fides_key is None:
-        disabled_resource_with_name = (
-            db.query(DataUseDbModel)
-            .filter(
-                DataUseDbModel.active.is_(False),
-                DataUseDbModel.name == data_use.name,
+    try:
+        if data_use.fides_key is None:
+            disabled_resource_with_name = (
+                db.query(DataUseDbModel)
+                .filter(
+                    DataUseDbModel.active.is_(False),
+                    DataUseDbModel.name == data_use.name,
+                )
+                .first()
             )
-            .first()
+            data_use.fides_key = get_key_from_data(
+                {"key": data_use.fides_key, "name": data_use.name}, DataUse.__name__
+            )
+            if data_use.parent_key:
+                data_use.fides_key = f"{data_use.parent_key}.{data_use.fides_key}"
+            if disabled_resource_with_name:
+                data_use.active = True
+                return disabled_resource_with_name.update(db, data=data_use.model_dump(mode="json"))  # type: ignore[union-attr]
+            return DataUseDbModel.create(db=db, data=data_use.model_dump(mode="json"))  # type: ignore[union-attr]
```

Comment:
```suggestion
```
We are calling this statement twice in. 145 and 146. can we consolidate by just leaving the one in 146?